src/Nethermind/Nethermind.Consensus/Producers/PayloadAttributes.cs

```diff
@@ -90,7 +90,20 @@ public static string ComputePayloadId(this PayloadAttributes payloadAttributes,
     }
 
     public static int GetVersion(this PayloadAttributes executionPayload) =>
-        executionPayload.Withdrawals is null ? 1 : 2;
+        executionPayload switch
+        {
+            { BeaconParentBlockRoot: not null, Withdrawals: not null } => EngineApiVersions.Cancun,
+            { Withdrawals: not null } => EngineApiVersions.Shanghai,
+            _ => EngineApiVersions.Paris
+        };
+
+    public static int ExpectedVersion(this IReleaseSpec spec) =>
```

Comment:
As this is on `IReleaseSpec` so can be seen broadly, maybe rename to `ExpectedEngineApiVersion`?